slang.h

```diff
@@ -1005,6 +1005,20 @@ extern "C"
 
         /** Clears any cached information */
         virtual SLANG_NO_THROW void SLANG_MCALL clearCache() = 0;
+
+        /** Write the data specified with data and size to the specified path.
+
+        Note that for normal slang operation it doesn't write files so this can return SLANG_E_NOT_IMPLEMENTED.
+
+        @param path The path for data to be saved to
+        @param data The data to be saved
+        @param size The size of the data
+        @returns SLANG_OK if successful (SLANG_E_NOT_IMPLEMENTED if not implemented, or some other error code)
+        */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL saveFile(
```

Comment:
At what point do we need to start claiming stability for these interfaces such that we can't go and add methods any more? For now this is a non-issue because nobody is implementing `ISlangFileSystemExt` outside of the Slang codebase itself, but it would be good if we had a metric to know when something like this is declared "stable."

Just to state the obvious (for the sake of any third party reading this review), us adding new methods to existing interfaces that Slang "exports" is not as much of an issue, and I only bring it up here because this is nominally an interface that Slang "imports" in the sense that client code might implement it and pass their implementation in, so that an old app running against a new Slang binary could crash when Slang tries to invoke a method that isn't there.

Reply:
Agreed this is an issue. That we have to be *more* careful with interfaces implemented in client code, than on exported interface. 

Some extra laxness is available with aspects of our interface that are exported c function. With interfaces it will require more discipline, and perhaps some other simplifying conventions over 'strict' COM.   

In this particular interface the method is sort of inevitable. I didn't initially add it because it would be a harder argument if it didn't have a clear and present usage. It does now. And although it could be implemented in other ways (say a different interface, or a callback mechanism or....) this is the natural place for it.